### PR TITLE
Octaneify MeSH

### PR DESCRIPTION
**Pull request by jrjohnson**
Had to do both <DetailMesh> and <MeshManager> at the same time as they
were too interrelated not to.

### CODE REVIEW

File: addon/components/mesh-manager.js [hunk 1]
```diff
@@ -1,120 +1,80 @@
...
-      return ProxiedDescriptors.create({
-        content: descriptor,
-        terms: terms
-      });
+
+    this.searching = false;
+    this.searchPage = 1;
+    this.hasMoreSearchResults = (descriptors.length > SEARCH_RESULTS_PER_PAGE);
+    if (this.hasMoreSearchResults) {
+      descriptors.pop();
```
Comment (stopfstedt) [inline addon/components/mesh-manager.js]: this blows up. descriptors is still and `DS.PromiseArray` object at this point, which subclasses `Ember.Array`, neither implement `pop()` . Please convert `descriptors` to a native JS array further up, using `toArray()`.

File: addon/components/mesh-manager.js [hunk 2]
```diff
@@ -1,120 +1,80 @@
...
+      offset: this.searchPage * SEARCH_RESULTS_PER_PAGE
     });
-    this.set('searchPage', this.get('searchPage') + 1);
-    this.set('hasMoreSearchResults', (results.length > this.get('searchResultsPerPage')));
-    if (this.get('hasMoreSearchResults')) {
-      results.pop();
+    this.searchPage = this.searchPage + 1;
+    this.hasMoreSearchResults = (descriptors.length > SEARCH_RESULTS_PER_PAGE);
+    if (this.hasMoreSearchResults) {
+      descriptors.pop();
```
Comment (stopfstedt) [inline addon/components/mesh-manager.js]: same thing, see comment about `pop()` above.

File: addon/components/mesh-manager.js (hunk 1 above)
Comment (jrjohnson) [reply addon/components/mesh-manager.js]: Fixed, added tests even!

File: addon/components/mesh-manager.js (hunk 2 above)
Comment (jrjohnson) [reply addon/components/mesh-manager.js]: Thanks for catching this, I thought me search for `Bees` was returning more than 50 results. Wrong!

**Pull request merged**